integration-test/background/storage.js

```diff
@@ -19,73 +17,111 @@ describe(`On https://${testPageDomain}/privacy-protections/storage-blocking/`, (
             },
             { polling: 100, timeout: 10000 }
         )
+        return { browser, page }
+    }
+
+    describe(`On https://${testPageDomain}/privacy-protections/storage-blocking/`, () => {
+        let cookies = []
 
-        let iframeFullyLoaded = false
-        try {
-            page.on('requestfinished', (req) => {
-                // once we see this url, we can consider the test completed
-                if (req.url().startsWith(`https://${thirdPartyTracker}/set-cookie`)) {
-                    iframeFullyLoaded = true
+        beforeAll(async () => {
+            let iframeFullyLoaded = false
+            const { browser, page } = await setup()
+            try {
+                page.on('requestfinished', (req) => {
+                    // once we see this url, we can consider the test completed
+                    if (req.url().startsWith(`https://${thirdPartyTracker}/set-cookie`)) {
+                        iframeFullyLoaded = true
+                    }
+                })
+                // Load the test pages home first to give some time for the extension background to start
+                // and register the content-script-message handler
+                await page.goto(`https://${testPageDomain}/`, { waitUntil: 'networkidle0' })
+                await page.bringToFront()
+                await page.goto(`https://${testPageDomain}/privacy-protections/storage-blocking/?store`, { waitUntil: 'networkidle0' })
+                await page.bringToFront()
+                // eslint-disable-next-line no-unmodified-loop-condition
+                while (!iframeFullyLoaded) {
+                    await wait.ms(100)
                 }
-            })
-            // Load the test pages home first to give some time for the extension background to start
-            // and register the content-script-message handler
-            await page.goto(`https://${testPageDomain}/`, { waitUntil: 'networkidle0' })
-            await page.goto(`https://${testPageDomain}/privacy-protections/storage-blocking/?store`, { waitUntil: 'networkidle0' })
-            // eslint-disable-next-line no-unmodified-loop-condition
-            while (!iframeFullyLoaded) {
-                await wait.ms(100)
+                // collect all browser cookies
+                do {
+                    await wait.ms(1000) // allow cookies to be set
+                    cookies = (await page._client.send('Network.getAllCookies')).cookies
+                } while (cookies.length === 0)
```

Comment:
do we want some safeguard like do this "max 5 times"?

Reply:
The test timeout is ~20s anyway, so we'd hit that cap not long after 5 times.